### PR TITLE
Improve errata test pos c-host search type

### PR DESCRIPTION
**Pull request by swadeley**
Hello

This test depends on updates to robottelo/constants.py in:
  Fix fake 9 security erratum constant #7806 

I will rebase when that is merged.

Edit: Done

Thank you

### CODE REVIEW

**Comment by swadeley** [comment]
Hello

I have rebased and pushed to this remote branch.

Test passes on 6.7, but 6.8 snap 3 fails on the RHBA check, its not there, its not just a search failure. Will consult devs to see if its a known issue or I should raise a bug.

robottelo]$ pytest  --show-capture=no tests/foreman/ui/test_errata.py -k test_positive_content_host_search_type
====================== test session starts ============================
platform linux -- Python 3.7.7, pytest-4.6.3, py-1.8.1, pluggy-0.13.1
shared_function enabled - OFF - scope:  - storage: file
rootdir: /sat6/robottelo
plugins: cov-2.8.1, services-1.3.1, mock-1.10.4, forked-1.1.3, xdist-1.31.0
collecting ... 2020-06-09 09:21:23 - conftest - DEBUG - Collected 13 test cases
2020-06-09 10:21:23 - robottelo.issue_handlers.bugzilla - DEBUG - Calling Bugzilla API for {'1655130'}
2020-06-09 10:21:25 - robottelo.ssh - DEBUG - Instantiated Paramiko client 0x7f587e227f50
2020-06-09 10:21:25 - robottelo.ssh - INFO - Connected to [dhcp-3-124.xxxx.redhat.com]
2020-06-09 10:21:25 - robottelo.ssh - INFO - >>> rpm -q satellite
2020-06-09 10:21:27 - robottelo.ssh - INFO - <<< stdout
satellite-6.7.1-1.el7sat.noarch

2020-06-09 10:21:27 - robottelo.ssh - DEBUG - Destroyed Paramiko client 0x7f587e227f50
2020-06-09 10:21:27 - robottelo.host_info - DEBUG - Host Satellite version: 6.7
2020-06-09 10:21:27 - robottelo.helpers - DEBUG - Generated file bz_cache.json with BZ collect data
collected 13 items / 12 deselected / 1 selected                                                                  

tests/foreman/ui/test_errata.py .                                                                          [100%]


=============== 1 passed, 12 deselected, 3 warnings in 507.75 seconds ==================

**Comment by swadeley** [comment]
Hello

this is now passing on Snap 4

robottelo]$ pytest tests/foreman/ui/test_errata.py -k test_positive_content_host_search_type
====================== test session starts =================================
platform linux -- Python 3.7.7, pytest-4.6.3, py-1.8.1, pluggy-0.13.1
shared_function enabled - OFF - scope:  - storage: file
rootdir: /sat6/robottelo
plugins: cov-2.8.1, services-1.3.1, mock-1.10.4, forked-1.1.3, xdist-1.31.0
collecting ... 2020-06-11 13:09:38 - conftest - DEBUG - Collected 13 test cases
2020-06-11 14:09:38 - robottelo.issue_handlers.bugzilla - DEBUG - Calling Bugzilla API for {'1655130'}
2020-06-11 14:09:40 - robottelo.ssh - DEBUG - Instantiated Paramiko client 0x7f59d80f2b10
2020-06-11 14:09:40 - robottelo.ssh - INFO - Connected to [dell-r330-12.xxxx.redhat.com]
2020-06-11 14:09:40 - robottelo.ssh - INFO - >>> rpm -q satellite
2020-06-11 14:09:42 - robottelo.ssh - INFO - <<< stdout
satellite-6.8.0-0.4.beta.el7sat.noarch

2020-06-11 14:09:42 - robottelo.ssh - DEBUG - Destroyed Paramiko client 0x7f59d80f2b10
2020-06-11 14:09:42 - robottelo.host_info - DEBUG - Host Satellite version: 6.8
2020-06-11 14:09:42 - robottelo.helpers - DEBUG - Generated file bz_cache.json with BZ collect data
collected 13 items / 12 deselected / 1 selected                                                                  

tests/foreman/ui/test_errata.py .                                                                          [100%]

============== 1 passed, 12 deselected, 3 warnings in 538.62 seconds ==================

**Comment by swadeley** [comment]
Hello @latran , you are secondary for this test's component, please read & review if possible.

Thank you

File: tests/foreman/ui/test_errata.py [hunk 1]
```diff
@@ -535,15 +542,39 @@ def test_positive_content_host_search_type(session, erratatype_vm):
...
+        # Search for RHBA buxfix errata
+        ch_erratum = session.contenthost.search_errata(
+            erratatype_vm.hostname, "type = bugfix", environment='Library Synced Content'
+        )
+
+        # Assert length matches known amount of RHBA errata
+        assert len(ch_erratum) == FAKE_10_YUM_BUGFIX_ERRATUM_COUNT
+
+        # Assert IDs are that of RHBA errata
+        errata_ids = sorted([e['Id'] for e in ch_erratum])
```
Comment (omkarkhatavkar) [inline tests/foreman/ui/test_errata.py:565]: instead of `e` we should give some appropriate name 'errata'

Comment (swadeley) [reply tests/foreman/ui/test_errata.py:565]: yep, can do (that is old code)

Comment (swadeley) [reply tests/foreman/ui/test_errata.py]: No with:         `errata_ids = sorted([erratum['Id'] for erratum in ch_erratum])`

**Comment by swadeley** [comment]
robottelo]$ pytest tests/foreman/ui/test_errata.py -k test_positive_content_host_search_type
================================== test session starts ===============================================
platform linux -- Python 3.7.7, pytest-4.6.3, py-1.8.1, pluggy-0.13.1
shared_function enabled - OFF - scope:  - storage: file
rootdir: /sat6/robottelo
plugins: cov-2.8.1, services-1.3.1, mock-1.10.4, forked-1.1.3, ipdb-0.1, xdist-1.31.0
collecting ... 2020-07-27 13:07:43 - conftest - DEBUG - Collected 13 test cases
2020-07-27 14:07:43 - robottelo.issue_handlers.bugzilla - DEBUG - Calling Bugzilla API for {'1655130'}
2020-07-27 14:07:45 - robottelo.ssh - DEBUG - Instantiated Paramiko client 0x7fb7a3dc6290
2020-07-27 14:07:45 - robottelo.ssh - INFO - Connected to [dhcp-3-134.xxxx.redhat.com]
2020-07-27 14:07:45 - robottelo.ssh - INFO - >>> rpm -q satellite
2020-07-27 14:07:47 - robottelo.ssh - INFO - <<< stdout
satellite-6.8.0-0.7.beta.el7sat.noarch

2020-07-27 14:07:47 - robottelo.ssh - DEBUG - Destroyed Paramiko client 0x7fb7a3dc6290
2020-07-27 14:07:47 - robottelo.host_info - DEBUG - Host Satellite version: 6.8
2020-07-27 14:07:47 - robottelo.helpers - DEBUG - Generated file bz_cache.json with BZ collect data
collected 13 items / 12 deselected / 1 selected                                                                  

tests/foreman/ui/test_errata.py .                                                                          [100%]


============================= 1 passed, 12 deselected, 3 warnings in 512.47 seconds =============

**Pull request merged**